rust/kernel/time.rs

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! This module contains timing methods for functionality like sleeping.
+
+use core::ffi::c_uint;
+
+use crate::bindings::msleep as msleep_bindgen;
+
+/// This method sleeps the current thread for `msecs` milliseconds, calling the
+/// kernel method msleep.
+pub fn msleep(msecs: c_uint) {
```

Comment:
Abstractions should not use `core::ffi::` or bindings' types. The idea is to provide abstractions, not "plain wrappers".

There are a few exceptions, like `from_kernel_errno`, but those have a good reason (e.g. it is intrinsic to their functionality).